### PR TITLE
OPS: Upgrade RN 0.64.2

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/eub8fdpkddqw42vn8hptrwef2c?device=pixel4

[download apk](http://filestorage.bluewallet.io:1488/rn64-1621307463.apk)

File: package.json [hunk 1]
```diff
@@ -65,6 +65,9 @@
     "transform": {
       "^.+\\.js$": "<rootDir>/node_modules/react-native/jest/preprocessor.js"
     },
+    "transformIgnorePatterns": [
```
Comment (marcosrdz) [inline package.json:68]: https://github.com/facebook/react-native/issues/31190

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/pm33dvpx3cnh00qfn1e7pp6qg8?device=pixel4

[download apk](http://filestorage.bluewallet.io:1488/rn64-1621308291.apk)

**Review by Overtorment — CHANGES_REQUESTED** [review]
```
> Task :react-native-haptic-feedback:extractDeepLinksDebug
/Users/runner/work/BlueWallet/BlueWallet/node_modules/react-native-haptic-feedback/android/build/intermediates/tmp/manifest/androidTest/debug/manifestMerger8350143561936519376.xml:5:5-74 Error:

	uses-sdk:minSdkVersion 16 cannot be smaller than version 21 declared in library [com.facebook.react:react-native:0.64.1] /Users/runner/.gradle/caches/transforms-2/files-2.1/c0dd7102f7e7bec9829c84a85a9ea345/jetified-react-native-0.64.1/AndroidManifest.xml as the library might be using APIs not available in 16
> Task :react-native-haptic-feedback:processDebugAndroidTestManifest FAILED
	Suggestion: use a compatible library with a minSdk of at most 16,
[androidx.vectordrawable:vectordrawable-animated:1.0.0] /Users/runner/.gradle/caches/transforms-2/files-2.1/98c7b0116a0584aebef186ec65cd8f39/vectordrawable-animated-1.0.0/AndroidManifest.xml Warning:
		or increase this project's minSdk version to at least 21,
	Package name 'androidx.vectordrawable' used in: androidx.vectordrawable:vectordrawable-animated:1.0.0, androidx.vectordrawable:vectordrawable:1.0.1.
		or use tools:overrideLibrary="com.facebook.react" to force usage (may lead to runtime failures)

```

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/xgrkpgqxnz181gfa3u9ec0xg7w?device=pixel4

[download apk](http://filestorage.bluewallet.io:1488/rn64-1621395363.apk)

**Comment by Overtorment** [comment]
e2e still fails

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/gvwx1ffyqf99bx71bzekn5fbzr?device=pixel4

[download apk](http://filestorage.bluewallet.io:1488/rn64-1622907235.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/n91fyyq3qup2w7btpwtf38hz2w?device=pixel4

[download apk](http://filestorage.bluewallet.io:1488/rn64-1622962635.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/dm4memzuhx78mhxmjp38fxte9w?device=pixel4

[download apk](http://filestorage.bluewallet.io:1488/rn64-1622973080.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/e0n949bx6bjbbveat3agcbv864?device=pixel4

[download apk](http://filestorage.bluewallet.io:1488/rn64-1622974809.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/bthfbj42zjpwye25p5kwafrm0g?device=pixel4

[download apk](http://filestorage.bluewallet.io:1488/rn64-1624034772.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/5yzmcqd5g25050vz524mg70z4m?device=pixel4

[download apk](http://206.189.62.190:1488/rn64-1624896510.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/4y8a0gwkw6cp5hav0zk3p69c0w?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7174)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/qmtqc5z9gy4jf3y02cy1tk4fnc?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7177)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/964degeex83h73bxbhqkbtaq3g?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7184)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/5t7am0kk5d1qxtveejaabt5evg?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7186)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/n4q4k6rmydmj7aw3kceg20ck5c?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7188)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/8u6qw6kkx89fhu7d2xhyt6wtv0?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7189)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/edegvjn08jt9fwu0c6q3pbkwvr?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7192)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/m3cyndx9f4rdw2td8muc83b3g4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7197)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/dbctkw6y2bq6r6x7ykrp5231e8?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7199)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/96wgnna80c107rd3t5nve49rng?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7200)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/b7gzmzfch1rc1nn66p8adgd1vm?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7201)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/gyb3uezqqy6nuavba77ykceu0m?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7202)

File: .flowconfig [hunk 2]
```diff
@@ -29,7 +29,7 @@ emoji=true
 
 esproposal.optional_chaining=enable
```
Comment (Overtorment) [inline .flowconfig:30]: we still cant remove flowconfig? we dont use flow

File: metro.config.js [hunk 3]
```diff
@@ -4,24 +4,26 @@
...
-      transform: {
-        experimentalImportSupport: false,
-        inlineRequires: false,
-      },
-    }),
+    //   blacklistRE: blacklist([
+    //     // This stops "react-native run-windows" from causing the metro server to crash if its already running
+    //     new RegExp(`${path.resolve(__dirname, 'windows').replace(/[/\\]/g, '/')}.*`),
+    //     // This prevents "react-native run-windows" from hitting: EBUSY: resource busy or locked, open msbuild.ProjectImports.zip
+    //     /.*\.ProjectImports\.zip/,
```
Comment (Overtorment) [inline metro.config.js:17]: dead code. candidate for removal next time you're refactoring around here

**Review by Overtorment — APPROVED** [review]
lgtm.

PS. sometimes i forget how shitty RN is under the hood
PPS. was jscore updated as well with this new RN version? i wonder if tor will stop hanging with this update

File: metro.config.js (hunk 3 above)
Comment (marcosrdz) [reply metro.config.js:17]: Need it for windows

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/rm17ufmvzgcg3jfv5758ye2k80?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7205)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/enxjemdx2dpcy255yvcd928bfg?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7206)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/5ckkd3cn6zfbhmxy3vxe817n5g?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7207)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/9hmt4410ugrcyekz054c099t9w?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7208)

File: .flowconfig (hunk 2 above)
Comment (marcosrdz) [reply .flowconfig:30]: Will check

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/2j1m459mpt3k6xkuhkv16tyj20?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7213)

**Comment by Overtorment** [comment]
@ncoelho @limpbrains last chance to weight in on this PR before I marge it. e2e issues seem to be resolved

**Comment by Overtorment** [comment]
@marcosrdz conflicts

**Comment by marcosrdz** [comment]
fixed

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/c5akqgg4gkfjj5hccbh7mkf19w?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7224)

**Pull request merged**